### DISCUSSION TITLE
Kanbans and Projects for the cold cases and ongoing uncategorized projects?

### DISCUSSION THREAD

**Discussion by balacij** [JacquesCarette/Drasil#3195] [Ideas]
For my thesis, Dr. Carette suggested we use kanbans to manage the leftover work, and I think it helped me get more focused on what needs to be done/what could be done later, but also helped my prioritize my tickets. Dr. Carette and I also briefly spoke about using kanbans more in the future, at the very least for my PhD.

However, I was wondering if we would be interested in using the GitHub Kanbans to also manage some of the existing projects in Drasil. This might be helpful for getting a quick glance at what the ongoing projects are in Drasil, what status they're in, and what they're blocked on. For the cold cases, this might help us to "unblock" some of the many that we haven't yet been able to close, or at least get an idea of what things are "hot topics" that should be worked out to unblock them. For the tickets that are largely blocked and have little content in their ticket/description, we could move them to "notes" in a related project/Kanban.

I know there are some current Projects and Kanbans but I don't think they've been updated in a while. Should we try to use them more? Are there any thoughts on it? Are there any better places than GitHub projects? JIRA is far more feature-rich, but I'm not sure it would integrate well with non-Atlassian products.

Right now, there are a few ongoing large projects, such as refactoring drasil-lang, adding more type checking, documentation, the website touch-ups, the repo touch-ups, etc. However, I'm not sure about the state of the old projects.

Pinging @JacquesCarette and @smiths in particular, of course, but all should contribute too!

**Comment by JacquesCarette** [comment]
Yes, absolutely, we should use as many of the tools that github provides as makes sense. The reason that the current projects fell into disuse is because the students involved "moved on" and there was no obvious person to take over.

We should stick to features that are well-integrated. We don't want to take on unnecessary technology risk.

The old projects should be either revived or closed. That should likely be done before new projects are created.

**Comment by smiths** [comment]
@samm82 was previously a user of GitHub projects.  If I remember correctly, he also used GitHub projects for other repos.  @samm82 did a nice tutorial for this year's capstone students on how to effectively use GitHub, and part of the tutorial touched on using GitHub projects.

There are other technologies for project management, but I think GitHub projects have everything we need.  They can't generate Gantt or Pert charts, but for a research project like Drasil, those charts wouldn't be particularly meaningful.

I like the idea of grouping related issues in a project.  In the recent issue from @samm82 (#3196) there are links to other related issues, but it takes time for the issue creator to search for related issues.  If all of the related content was together in a project board, the related information would be in one place.

I'm planning to create issues related to improving the Projectile example.  I might experiment with putting these issues together as part of a project.  :smile:

**Comment by samm82** [comment]
@smiths Note that there is already a [Projectile project](https://github.com/JacquesCarette/Drasil/projects/5)! Creating a project for the chunk investigation to keep things organized is a good idea though!

**Reply by smiths** [reply]
Great!  I'll make a point of adding any projectile related issues that I create to this project.

**Comment by balacij** [comment]
Perfect! :) 

Regarding the projects, are we able to create new-style Project boards and link them to Drasil? It seems they're tied to user accounts or organization rather than repos, unlike the classic-style boards. So, when I create a board, it creates it for _my_ github account, and I can't link it to Drasil.

Right now, I'd like to create boards for:
1. Continuing the typing issues and typing GOOL
2. Merging/implementing the ChunkDB prototype
3. Refactoring some areas of code based on #2883 (e.g., #2885, #2886, #2887)

**Reply by JacquesCarette** [reply]
I don't understand the design of new-style projects, and the rationale to tie them to accounts (or organizations). Create old-style boards for now?

**Reply by smiths** [reply]
Unfortunately, I don't think the old-style boards are an option.  For some reason, GitHub has switched to organization boards.

**Reply by samm82** [reply]
By selecting "Projects (classic)", you can still create an old-style board!

![image](https://user-images.githubusercontent.com/35857611/209441735-539c781e-b068-48db-a31d-a3a0726cfceb.png)
![image](https://user-images.githubusercontent.com/35857611/209441745-4ba6ea06-593d-49a4-98fe-9de397c83499.png)

**Reply by smiths** [reply]
Great!  I didn't know that.  Thanks @samm82.

**Reply by balacij** [reply]
I think the rationale for tying projects to accounts (or organizations) instead of repositories is that some projects span across more than one repository. For example, a website split up in a `frontend` repo and a `backend` repo, which also might have different user permissions across them.

**Reply by balacij** [reply]
Using old style projects sounds fine. If we ever switch to multiple repositories, I'm sure we'll benefit from switching to a "Drasil" organization and using the new style projects.
